nixos/doc/manual/configuration/linux-kernel.chapter.md

```diff
@@ -138,3 +148,23 @@ $ cd linux-*
 $ make -C $dev/lib/modules/*/build M=$(pwd)/drivers/net/ethernet/mellanox modules
 # insmod ./drivers/net/ethernet/mellanox/mlx5/core/mlx5_core.ko
 ```
+
+## ZFS {#sec-linux-zfs}
+
+It's a common issue that the latest stable version of ZFS doesn't support the latest
+available Linux kernel. It's possible to pin the system to the latest available kernel
+version *that is supported by ZFS* like this:
+
+```nix
+{
+  boot.kernelPackages = pkgs.zfs.latestCompatibleLinuxPackages;
```

Comment:
Thinking about this a bit more...

Given that the manual makes no mention of ZFS, and given the issues that using `latestCompatibleLinuxPackages`, should this really be the first option that we give to users? Imagine a novice user that googles "nixos zfs docs" and this is the first result that pops up (and the only one under `https://nixos.org/manual/nixos/`). Many people would copy this code snippet and not bother reading the warnings that follow. Could we instead suggest something safer, along the lines of:

```suggestion
  
  # Use the current longterm kernel, if ZFS supports it,
  # otherwise use an older one that is the latest supported by ZFS.
  boot.kernelPackages = let
    defaultKernel = pkgs.linuxPackages.kernel
    zfsLatestKernel = pkgs.zfs.latestCompatibleLinuxPackages.kernel
  in
    if zfsLatestKernel.kernelAtLeast defaultKernel.version
    then defaultKernel
    else zfsLatestKernel;
```

Reply:
We should not tell people to copy this into their configuration and rather move it to the zfs packages as latestCompatibleLTSLinuxPackages.